framework/shader_compiler.h

```diff
@@ -54,19 +54,28 @@ class GLSLCompiler
 	static void reset_target_environment();
 
 	/**
-	 * @brief Compiles GLSL to SPIRV code
+	 * @brief Compiles GLSL/HLSL to SPIRV code
 	 * @param stage The Vulkan shader stage flag
-	 * @param glsl_source The GLSL source code to be compiled
+	 * @param src_language The language of source code
+	 * @param shader_source The source code to be compiled
 	 * @param entry_point The entrypoint function name of the shader stage
 	 * @param shader_variant The shader variant
 	 * @param[out] spirv The generated SPIRV code
 	 * @param[out] info_log Stores any log messages during the compilation process
 	 */
 	bool compile_to_spirv(VkShaderStageFlagBits       stage,
-	                      const std::vector<uint8_t> &glsl_source,
-	                      const std::string &         entry_point,
-	                      const ShaderVariant &       shader_variant,
+	                      vkb::ShaderSourceLanguage   src_language,
+	                      const std::vector<uint8_t> &shader_source,
+	                      const std::string          &entry_point,
+	                      const ShaderVariant        &shader_variant,
 	                      std::vector<std::uint32_t> &spirv,
-	                      std::string &               info_log);
+	                      std::string                &info_log);
+
+	bool compile_to_spirv(VkShaderStageFlagBits       stage,
+	                      const std::vector<uint8_t> &shader_source,
+	                      const std::string          &entry_point,
+	                      const ShaderVariant        &shader_variant,
+	                      std::vector<std::uint32_t> &spirv,
+	                      std::string                &info_log);
```

Comment:
Instead of having two flavours of `compile_to_spirv` I would prefer to have just one. Maybe with some default value for the `src_language`.
And please extend the HPPShaderCompiler accordingly.